src/dapp/walletProvider.ts

```diff
@@ -289,6 +291,84 @@ export class WalletProvider implements IDappProvider {
         });
     }
 
+    /**
+     * Packs a {@link Transaction} and fetches correct redirect URL from the wallet API. Then redirects
+     *   the client to the sign transaction hook
+     * @param transaction
+     * @param options
+     */
+    async signTransaction(transaction: Transaction, options?: { callbackUrl?: string }): Promise<Transaction> {
+        if (!this.mainFrame) {
+            throw new Error("Wallet provider is not initialised, call init() first");
+        }
+
+        const { contentWindow } = this.mainFrame;
+        if (!contentWindow) {
+            throw new Error("Wallet provider is not initialised, call init() first");
+        }
+
+        return new Promise<Transaction>((resolve, reject) => {
+            let plainTransaction = WalletProvider.prepareWalletTransaction(transaction);
+            console.log("postMessage", DAPP_MESSAGE_SIGN_TRANSACTION_URL, plainTransaction);
+
+            contentWindow.postMessage({
+                type: DAPP_MESSAGE_SIGN_TRANSACTION_URL,
+                data: {
+                    transaction: plainTransaction
+                }
+            }, this.walletUrl);
+
+            const timeout = setTimeout(_ => reject('sign transaction url not responding'), 5000);
+            const signTransactionUrl = (ev: IDappMessageEvent) => {
+                console.log("event", "signTransactionUrl", ev);
+
+                if (!this.isValidWalletSource(ev.origin)) {
+                    return;
+                }
+
+                const { data } = ev;
+                if (data.type !== DAPP_MESSAGE_SIGN_TRANSACTION_URL) {
+                    return;
+                }
```

Comment:
Should these be `return reject(...);` instead?

Reply:
Same as above - these were changes from the development branch.